src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -392,7 +392,26 @@ sparqlExpression::IdOrLiteralOrIri IriOrUriValueGetter::operator()(
 //______________________________________________________________________________
 std::optional<std::string> LanguageTagValueGetter::operator()(
     ValueId id, const EvaluationContext* context) const {
-  return getValue<std::optional<std::string>>(id, context, *this);
+  using enum Datatype;
+  switch (id.getDatatype()) {
+    case Bool:
+    case Int:
+    case Double:
+    case Date:
+    case GeoPoint:
+      // For literals without language tag, we return an empty string per
+      // standard.
+      return {""};
+    case Undefined:
+    case VocabIndex:
+    case LocalVocabIndex:
+    case TextRecordIndex:
+    case WordVocabIndex:
+    case BlankNodeIndex:
+      return getValue<std::optional<std::string>>(id, context, *this);
+    default:
+      AD_FAIL();
+  }
```

Comment:
What happens if you drop the `default` and move the `AD_FAIL()` after the switch?
This pattern has the advantage that compilers will warn once we add an additional enum value.